### ISSUE TITLE
when updating to 0.4 using 3-col homepage, widget area is unregistered

### ISSUE THREAD

**Issue by aschweigert** [INN/Largo#375]
The left column widget area is empty post-update (the widgets still show up in the WP admin under the "inactive sidebars" section, it's just not matched up to the homepage left rail widget area.

**Comment by willhaynes** [comment]
Tested updating with the 3col layout enabled the various configurations:

 - `v0.3 → v0.4`
 - `v0.3 → v0.4.1`
 - `v0.3 → v0.4 → v0.4.1`

And was not able to replicate what was happening here. The names of the sidebar did not change in either instance, so it's not obvious what would introduce it being unregistered.

Are there any more recent examples?

**Comment by aschweigert** [comment]
might be fixed by this point. @rnagle do you want to double check this one and close if we're not able to replicate?

**Comment by rnagle** [comment]
I *think* there might be a scenario where this still happens based on this snippet: https://github.com/INN/Largo/blob/master/inc/sidebars.php#L72-L78

After updating, the `homepage_layout` option is replaced by `home_template`. So, `homepage_layout` may still be set and may be something other than `3col` as required by the condition in the code snippet referenced above.

So, I think what we need to do is change that condition to do something like:

    if (of_get_option('home_template') == 'LegacyThreeColumn') { ...

@willhaynes What do you think? It might be worth testing before closing this one.

See also: https://github.com/INN/Largo/blob/master/inc/update.php#L82-L83

**Comment by willhaynes** [comment]
As far as I can tell, when upgrading `v0.3 → v0.4` the `homepage_layout` option is totally eliminated (I took [an option dump immediately before & after updating __here__](https://gist.github.com/willhaynes/53ef7c41de0386369490)).

So yes, [the conditional here](https://github.com/INN/Largo/blob/master/inc/sidebars.php#L72) will never be true post update, and the "homepage left rail" sidebar is not registered there.

But the sidebar isn't lost, instead [it's registered in the homepage class directly](https://github.com/INN/Largo/blob/master/homepages/layouts/LegacyThreeColumn.php).

**Comment by rnagle** [comment]
OK -- so, the conditional I point to shouldn't present any issue as long as the site on `0.3` is using the three column layout, since it will be migrated to the new option name and settings for `0.4`. This seems to work fine based on @willhaynes's report and my experience testing this.

However, I did notice something that leads a different direction. When we upgrade sites from `0.3` to `0.4`, we are changing the child theme to point at `largo-dev`, since the old and new versions of Largo are required for our deployment.

To get the upgraded site to actually pick up on the transition to `0.4`, we have to deactivate and reactivate the child theme. I've found that if I switch from `0.3` theme to one of the WP Twenty * themes and reactivate the child theme, the sidebars are unassigned.

My guess is that this happens because all of the registered Largo widgets go away when switching the Twenty * theme.

I think that this issue can be closed, pending removal of some unnecessary code related to the sidebars registered for the three column homepage layout. We should open an internal admin ticket to discuss the upgrade/update process and figure out how to deal with the transition in a more graceful manner.
